langchain4j-watsonx/src/test/java/dev/langchain4j/model/watsonx/WatsonxChatModelIT.java

```
@@ -0,0 +1,111 @@
+import org.junit.jupiter.api.Disabled;
+import org.junit.jupiter.api.condition.EnabledIfEnvironmentVariable;
+
+@EnabledIfEnvironmentVariable(named = "WATSONX_API_KEY", matches = ".+")
+@EnabledIfEnvironmentVariable(named = "WATSONX_PROJECT_ID", matches = ".+")
+@EnabledIfEnvironmentVariable(named = "WATSONX_URL", matches = ".+")
+public class WatsonxChatModelIT extends AbstractChatModelIT {
+
+    static final String API_KEY = System.getenv("WATSONX_API_KEY");
+    static final String PROJECT_ID = System.getenv("WATSONX_PROJECT_ID");
+    static final String URL = System.getenv("WATSONX_URL");
+
+    static final AuthenticationProvider authProvider =
+            IAMAuthenticator.builder().apiKey(API_KEY).build();
+
+    @Override
+    protected List<ChatModel> models() {
+        return List.of(WatsonxChatModel.builder()
+                .service(createChatService("meta-llama/llama-4-maverick-17b-128e-instruct-fp8"))
+                .enableJsonSchema(true)
+                .build());
+    }
+
+    @Override
+    protected String customModelName() {
+        return "ibm/granite-3-3-8b-instruct";
+    }
+
+    @Override
+    protected ChatModel createModelWith(ChatRequestParameters parameters) {
+        return WatsonxChatModel.builder()
+                .service(createChatService("ibm/granite-3-3-8b-instruct"))
+                .defaultRequestParameters(parameters)
+                .build();
+    }
+
+    @Override
+    protected ChatRequestParameters createIntegrationSpecificParameters(int maxOutputTokens) {
+        return WatsonxChatRequestParameters.builder()
+                .maxOutputTokens(maxOutputTokens)
+                .build();
+    }
+
+    @Override
+    public boolean supportsSingleImageInputAsPublicURL() {
+        // Watsonx does not support images as URLs, only as Base64-encoded strings
+        return false;
+    }
+
+    @Override
+    @Disabled("It seems that supportsSingleImageInputAsPublicURL doesn't work with this method")
```

What does it mean? `WatsonxChatModel` should throw `UnsupportedFeatureException` if image is specified as URL